ENH: Add to_latex() method to Series (#16180)

bsweger opened the pull request:
This changeset adds _repr_latex_ to the Series class and moves
the to_latex() method from the DataFrame class to the NDFrame
class.

 - [ ] closes #16180
 - [ ] tests added / passed
 - [ ] passes ``git diff upstream/master --name-only -- '*.py' | flake8 --diff``
 - [ ] whatsnew entry

Commit:
ENH: Add to_latex() method to Series (#16180)

This changeset adds _repr_latex_ to the Series class and moves
the to_latex() method from the DataFrame class to the NDFrame
class.

TomAugspurger commented on pandas/core/generic.py:
```diff
@@ -1502,6 +1502,95 @@ def to_xarray(self):
+                 column_format=None, longtable=None, escape=None,
+                 encoding=None, decimal='.', multicolumn=None,
+                 multicolumn_format=None, multirow=None):
+        # Get defaults from the pandas config
```
I think you'll need to start this with

```python
if self.ndim == 1:
    self = self.to_frame()
```

then the rest of it should be ok.

bsweger replied:
Thanks for this suggestion--doing the check here rather than invoking `to_frame()` in the Series _repr_latex method makes a lot of sense.

TomAugspurger commented on pandas/core/series.py:
```diff
@@ -391,6 +391,16 @@ def get_values(self):
         """ same as values (but handles sparseness conversions); is a view """
         return self._data.get_values()
 
+    def _repr_latex_(self):
```
Can you actually remove this and move the `_repr_latex_` from `pandas/core/frame.py` and to `NDFrame` in `pandas/core/generic.py`?

TomAugspurger reviewed (changes requested):
Sorry I missed this yesterday. A couple of comments.

Could you also add some tests for `Series.to_latex` directly (keep the one testing `_repr_latex_` though. They can go in `pandas/tests/io/formats/test_to_latex.py`

Thanks!

Commit:
Add Series to_latex test

Commit:
Move _repr_latex_ to NDFrame

Streamline things a bit by moving _repr_latex_ methods out of
the Series and DataFrame classes

bsweger commented:
@TomAugspurger Thanks for the review! Moved things around per your suggestions and added a test explicitly for `Series.to_latex()`. It's a very simple test to make sure the function works; wasn't sure if the entire suite of dataframe-related latex test options was worth repeating for a series. Happy to add more if you think it's helpful.

TomAugspurger committed:
DOC: Added versionchanged

TomAugspurger commented:
@bsweger this looks fantastic. I'm pushing a tiny doc change that I forgot to ask for (adding a note that it was added to series in 0.20.2), then merging.
